eclipse-scout-core/src/util/ObjectUuidProvider.ts

```diff
@@ -0,0 +1,326 @@
+/*
+ * Copyright (c) 2010, 2025 BSI Business Systems Integration AG
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+import {Constructor, numbers, ObjectFactory, ObjectModel, objects, ObjectWithType, ObjectWithUuid, scout, SomeRequired, strings, Widget} from '../index';
+
+/**
+ * Helper class to extract IDs of objects and to compute uuidPaths.
+ */
+export class ObjectUuidProvider implements ObjectUuidProviderModel, ObjectWithType {
+
+  declare model: ObjectUuidProviderModel;
+  declare initModel: SomeRequired<this['model'], 'object'>;
+  declare self: ObjectUuidProvider;
+
+  objectType: string;
+  id: string;
+
+  protected _uiSeqIdNo = 0;
```

Comment:
Maybe remove the "No" suffix? Or keep it but remove the "Id" bit instead? Currently, the same seems a bit weird.